src/QMCHamiltonians/ECPComponentBuilder.cpp

```diff
@@ -207,6 +207,43 @@ bool ECPComponentBuilder::put(xmlNodePtr cur)
   }
   if (pp_nonloc)
   {
+    if (Nrule == -1)
+    {
+      //Sperical quadrature rules set by exact integration up to lmax of
+      //nonlocal channels.
+      //From J. Chem. Phys. 95 (3467) (1991)
+      switch (pp_nonloc->lmax)
+      {
+      case 0:
+        Nrule = 1;
+        break;
+      case 1:
+      case 2:
+        Nrule = 2;
```

Comment:
I recommend we keep a floor at nrule=4.  I'm pretty sure some d-local potentials have higher variance at these lower settings.

Reply:
Also, we don't want to degrade the statistical performance of existing inputs.

Reply:
changed the floor to nrule=4